src/com/amazon/corretto/crypto/provider/HkdfRfc5869SecretKeyFactorySpi.java

```diff
@@ -0,0 +1,143 @@
+            resultBytes.length,
+            digestCode,
+            spec.secretOrPrk,
+            spec.secretOrPrk.length,
+            spec.salt,
+            spec.salt.length,
+            spec.info,
+            spec.info.length);
+        break;
+      case HKDF_EXTRACT_MODE:
+        // In extract mode, we ignore the value of desiredLength and use
+        // the output size of the digest as desiredLength
+        resultBytes = new byte[digestLength];
+        hkdfExtract(
+            resultBytes,
+            resultBytes.length,
+            digestCode,
+            spec.secretOrPrk,
+            spec.secretOrPrk.length,
+            spec.salt,
+            spec.salt.length);
+        break;
+      default:
+        // spec.mode is HKDF_EXPAND_MODE. We already checked for it in the constructor of
+        // HkdfRfc5869Spec
+        resultBytes = new byte[spec.desiredLength];
+        hkdfExpand(
+            resultBytes,
+            resultBytes.length,
+            digestCode,
+            spec.secretOrPrk,
+            spec.secretOrPrk.length,
+            spec.info,
+            spec.info.length);
+    }
+
+    return new SecretKeySpec(resultBytes, spec.algorithmName);
+  }
+
+  private static native void hkdf(
+      byte[] jOutput,
+      int outputLen,
+      int digestCode,
+      byte[] jSecret,
+      int secretLen,
+      byte[] jSalt,
+      int saltLen,
+      byte[] jInfo,
+      int infoLen);
+
+  private static native void hkdfExtract(
+      byte[] jOutput,
+      int outputLen,
+      int digestCode,
+      byte[] jSecret,
+      int secretLen,
+      byte[] jSalt,
+      int saltLen);
+
+  private static native void hkdfExpand(
+      byte[] jOutput,
+      int outputLen,
+      int digestCode,
+      byte[] jPrk,
+      int prkLen,
+      byte[] jInfo,
+      int infoLen);
+
+  private static native int getDigestLen(int digestCode);
+
+  @Override
+  protected KeySpec engineGetKeySpec(final SecretKey key, final Class<?> keySpec) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  protected SecretKey engineTranslateKey(final SecretKey key) {
+    throw new UnsupportedOperationException();
+  }
+
+  static final Map<String, HkdfRfc5869SecretKeyFactorySpi> INSTANCES = getInstances();
+
+  private static final String HKDF_RFC5869 = "HkdfRfc5869";
+  private static final String WITH = "With";
+  static final String HKDF_RFC5869_WITH_SHA1 = HKDF_RFC5869 + WITH + "SHA-1";
+  static final String HKDF_RFC5869_WITH_SHA256 = HKDF_RFC5869 + WITH + "SHA-256";
+  static final String HKDF_RFC5869_WITH_SHA384 = HKDF_RFC5869 + WITH + "SHA-384";
+  static final String HKDF_RFC5869_WITH_SHA512 = HKDF_RFC5869 + WITH + "SHA-512";
+
+  private static Map<String, HkdfRfc5869SecretKeyFactorySpi> getInstances() {
+    final Map<String, HkdfRfc5869SecretKeyFactorySpi> result = new HashMap<>();
+    result.put(
+        HKDF_RFC5869_WITH_SHA1.toUpperCase(),
+        new HkdfRfc5869SecretKeyFactorySpi(1, getDigestLen(1)));
```

Comment:
These digest codes seem somewhat arbitrary, it'd be more readable to define some kind of aliasing for them.

Reply:
Done.